### PR TITLE
added cloud functions demo with raw geoJSON text

### PR DESCRIPTION
**Pull request by mysticetii**
Functions demo where getGeoJSON returns randomly a geoJSON or error + fixed firestore in web bug

Cloud function that returns the call:

```
export const getGeoJSON = functions.https.onCall((data, context) => {
  const randomNumber = Math.random();
  const coordinates = [[-0.10773, 51.51423], [-0.08982, 51.51748]];
  const geoJSONData = {
    "type": "Feature",
    "geometry": {
      "type": "MultiPoint",
      "coordinates": coordinates,
    },
    "properties": {
      "name": "Points in London",
    },
  };
  if (randomNumber >= 0.3) {
    console.log(JSON.stringify(geoJSONData));
    return (JSON.stringify(geoJSONData));
  } else {
    throw new HttpsError("not-found", "Random number lower than 30%.");
  }
});
```

### CODE REVIEW

**Review by joykuotw — CHANGES_REQUESTED** [review]

**Comment by joykuotw** [comment]
The GeoJSON format is incorrect. Please update it.

File: lib/firebase_functions/firebase_functions_page.dart [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      context: context,
+      builder: (context) => alert,
+    );
+  }
+
+  @override
+  void initState() {
+    functions = FirebaseFunctions.instance;
+    super.initState();
+  }
```
[the remark this answers is not in the document]
Comment (mysticetii) [reply lib/firebase_functions/firebase_functions_page.dart:47]: Sorry. functions variable will be used in the next commit in the function call.

File: pubspec.yaml [hunk 2]
```diff
@@ -33,13 +33,15 @@ dependencies:
...
   cloud_firestore: ^1.0.4
+  cloud_functions: ^1.0.3
   modal_progress_hud: ^0.1.3
   provider: ^5.0.0
   camera: ^0.8.1
   video_player: ^2.0.2
   path_provider: ^2.0.1
   date_format: ^2.0.2
   geolocator: ^7.0.1
+  pretty_json: ^2.0.0
```
[the remark this answers is not in the document]
Comment (mysticetii) [reply pubspec.yaml:44]: Removed the package.

**Comment by mysticetii** [comment]
Updated the GeoJSON data return.

**Review by joykuotw — APPROVED** [review]

**Comment by joykuotw** [comment]
we might have to provide the deployed function code in our codebase for instruction purpose.

**Pull request merged**